force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -163,126 +196,164 @@ public with sharing class ClassesObjectsMethods {
     }
 
-    /**
-     * Question 9
-     * Given a date time return the day of the week in number format
-     * @param dt the current date and time
-     * @return The number of the day of the week for the given date.
-     */
+    // /**
+    //  * Question 9
+    //  * Given a date time return the day of the week in number format
+    //  * @param dt the current date and time
+    //  * @return The number of the day of the week for the given date.
+    //  */
     public static String getDayOfWeek(DateTime dt) {
-        // Define a map of day of the week as a key and day number as the value
-        // Monday = 1, Tuesday = 2, etc.
-
-        // Get the day name using .format('EEEE');
-
-        // Get the day number from the map using the day name
-
-        // Return the result as a string
-        return null; // Replace null with the variable you used to store the result
+      
+      
+       String dayOfTheWeek = dt.format('EEEE');
+       System.debug(dayOfTheWeek);
+       Map<String, Integer> dayMap = new Map<String, Integer>{
+           'Monday' => 1,
+           'Tuesday' => 2,
+           'Wednesday' => 3,
+           'Thursday' => 4,
+           'Friday' => 5,
+           'Saturday' => 6,
+           'Sunday' => 7};
+       Integer NumberOfWeek = dayMap.get(dayOfTheWeek);
+       return String.valueOf(NumberOfWeek);
+
+    //     // Monday = 1, Tuesday = 2, etc.
+
+    //     // Get the day name using .format('EEEE');
+
+    //     // Get the day number from the map using the day name
+
+    //     // Return the result as a string; // Replace null with the variable you used to store the result
     }
 
 
-    /**
-     * Question 10
-     * Calculate the difference between the years of two dates
-     * For example, if the first date is 2023-01-01 and the second date is 2025-12-31, the method should return 2
-     * If either of the dates is null, return null.
-     * @param d1 The first date.
-     * @param d2 The second date.
-     * @return The a positive number years between the two dates, or null if either date is null.
-     */
+    // /**
+    //  * Question 10
+    //  * Calculate the difference between the years of two dates
+    //  * For example, if the first date is 2023-01-01 and the second date is 2025-12-31, the method should return 2
+    //  * If either of the dates is null, return null.
+    //  * @param d1 The first date.
+    //  * @param d2 The second date.
+    //  * @return The a positive number years between the two dates, or null if either date is null.
+    //  */
     public static Integer calculateYearDifference(Date d1, Date d2) {
-
-        // Subtract the difference in years
-
-        return null; // Replace null with the variable you used to store the result
+        Integer Yeardifference = null;
+        if (d1 !=null && d2!=null) {
+        Yeardifference = Math.abs(d2.year() - d1.year());
+        } 
+        return Yeardifference;
```

Comment:
good. This logic fine. I  find that my code seems more readable if I do the null checks  first. And  leave my  the main code branch out of that null check. 
```
  public static Integer calculateYearDifference(Date d1, Date d2) {
        if (d1 == null || d2 == null) {
            return null;
        }
        return Math.abs(d1.year() - d2.year());
    }
    ```